src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,57 @@
+/* eslint-disable no-useless-constructor */
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  static removeDead(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden = false, health = 100) {
+    super(name, health);
+    this.hidden = hidden;
+
+    if (this.health <= 0) {
+      Animal.removeDead(this);
+    }
```

Comment:
Calling `Animal.removeDead(this)` in the constructor of `Herbivore` might not be necessary. A newly created herbivore should not be dead, so this check could be redundant.